discover/amd_common.go

```diff
@@ -47,7 +47,11 @@ func rocmGetVisibleDevicesEnv(gpuInfo []GpuInfo) (string, string) {
 		}
 		ids = append(ids, info.ID)
 	}
-	return "HIP_VISIBLE_DEVICES", strings.Join(ids, ",")
+	// There are 3 potential env vars to use to select GPUs.
+	// ROCR_VISIBLE_DEVICES supports UUID or numeric so is our preferred
+	// GPU_DEVICE_ORDINAL supports numeric IDs only
+	// HIP_VISIBLE_DEVICES supports numeric IDs only
+	return "ROCR_VISIBLE_DEVICES", strings.Join(ids, ",")
```

Comment:
Does this cause any backwards compatibility issues for people who are currently setting `HIP_VISIBLE_DEVICES`?

Reply:
Good question.  Let me test to see what the behavior is.

Reply:
It's not ideal.  It winds up masking all the GPUs incorrectly.  Let me clean up that case before we merge this.

Reply:
fixed